Scaffold Refactors

jonahtwalker opened the pull request:
<!-- This template is to serve as a comprehensive outline for documenting updates to the code. Some of these sections may not be relevant to your PR. In that case, document it inline with `N/A`  -->

## What does this PR do?
<!-- Provide a clear and concise description of what this PR accomplishes -->
This contains refactors to use more update to date Svelte5 conventions wherever possible, and to realign with the component library we've already started to overwrite. 

- [x] Refactored SideBar
- [x] Update the SideBar button #41 
- [x] Refactored UserNav #40
- [x] Refactored Footer

### Related Issues & PRs
<!-- List the related issues, using either: `Closes #123` or `Fixes #123` , which will automatically move the specified issue to "In Review" when the PR is created -->
#34
#25
#32

Fixes #41

#### Key Decisions
<!-- Explain the technical decisions and alternative approaches considered for your new implementations and refactored solutions -->
- **Implemeneted Approach**:
  <!-- What did you decide on? -->
- **Alternatives Considered**:
  <!-- What else did you consider? -->
- **Rationale**:
  <!-- Why did you decide this was the most appropriate solution? -->

## Breaking Changes
<!-- Do others need to do something to successfully incorporate your changes into their work? -->
- [ ] This PR includes breaking changes
  <!-- What do they need to do? -->

### Limitations & Concerns
<!-- Describe any limitations, potential edge cases, or situations where your solution might break -->
- **Known limitations**:
- **Edge cases**:
- **Potential risks**:

---

### UI Changes
<!-- If this PR modifies the UI, provide screenshots and descriptions -->
- [x] This PR includes UI changes
<!-- Include screenshots in the table below, include before & after applicable -->

### Screenshots
| Description | Screenshot |
|-------------|------------|
| Before      | <!-- img(s) --> |
| After       | ![Screenshot 2025-03-12 at 6 18 14 PM](https://github.com/user-attachments/assets/ef8fac6f-bbec-4d4c-8c3c-67f3dc0a28d8)  ![Screenshot 2025-03-12 at 6 17 55 PM](https://github.com/user-attachments/assets/3c28d795-8fed-480f-9a52-78fdabbd6468) |
| ![Screenshot 2025-03-17 at 2 34 42 PM](https://github.com/user-attachments/assets/d49a6236-1f55-450b-994c-7ee8068f0784) | ![Screenshot 2025-03-17 at 2 34 24 PM](https://github.com/user-attachments/assets/831a7e17-51ed-423a-8ccc-87345163a5af) |

---

### Next Steps
<!-- What needs to happen after this PR is merged? -->
- [ ] Follow-up tasks are created
  - [ ] Task 1

### How to Test
<!-- Provide detailed steps for QA to test the changes -->
1. Prerequisites:
  <!-- e.g. Do I need to be a specific role? -->
There are no preliminary steps to test any of these changes.
2. Test steps:
   <!-- What is the quickest path to testing your solution? -->
3. Expected results:
   <!-- Is there any nuance that should be clarified? If so, can the implementation get more simple instead? -->

## Final Checklist
- [x] Linked related issues
- [x] Rebased with target branch
- [x] No lint errors
- [x] QA'd in multiple browsers
- [x] Added/updated tests (if applicable)
- [x] Mobile responsive (if applicable)
- [ ] Updated documentation (if applicable)
- [ ] Notified the Team of this updates readiness

---

## AI Usage
<!-- Document any AI tools you used to develop this -->
- [ ] AI tools were used in development
  - **Tool used**:
    <!-- Specify any model used -->
  - **Prompts used**:
    <!-- Format these inclusions as quotes, like: `> Provide  -->
    - **Initial prompt**: 
- [x] No AI tools were used
  - **Reason**:
    <!-- Why not? -->
    The shadcn-svelte component library has numerous examples of Svelte5 code to reference.

## References
<!-- List any helpful references used during development -->
- **Documentation**:
[Example Sidebar](https://github.com/huntabyte/shadcn-svelte/blob/next/sites/docs/src/lib/registry/new-york/block/sidebar-09/components/app-sidebar.svelte)
- **Discussions**:
- **Related PRs**:

jonahtwalker committed:
- refactored Sidebar to use Svelte5 conventions

jonahtwalker committed:
- Reverted SidebarTrigger changes and updated reactive icon
- More cleanup in AppSideBar

jonahtwalker committed:
Reconcile AppSidebar

jonahtwalker committed:
- updated shad Dialog
- removed extra docs dir
- refactored settings page and tabs
- added null check fix for unit tests to SideBar trigger icon
- refactored the password validation logic into a util file
- removed unnecessary awaits from tests
- removed the svelte llm instructions in favor of the medium svelte5 doc in `docs`

jonahtwalker commented on docs/faq-data.ts:
This file was moved from `lib/docs` to `docs`.

rupalirajput2597 replied:
Cool, anyways this file just have static FAQs.

jonahtwalker commented on src/lib/components/app-sidebar.svelte:
```diff
@@ -42,70 +32,58 @@
-    </Sidebar.Content>
-  </Sidebar.Root>
-</div>
+            <span class="font-lexend truncate ps-2 text-xl font-bold text-primary">First Spark</span
```
`truncate` was the tailwind class that is necessary for the First Spark text to collapse to the Logo icon as the other menu items do.

rupalirajput2597 replied:
Thanks for the clarification Jonah!

jonahtwalker commented on src/lib/components/app-sidebar.svelte:
```diff
@@ -42,70 +32,58 @@
-          {/if}
-        </a>
-      </div>
+  let {
```
Maintaining this standard set by the shadcn components and examples is important so that it is easier to extend the functionality of the components and blocks very similarly throughout our entire component library.

Let's make it a habit of checking for updates before we totally customize what are meant to be ready-made components.

rupalirajput2597 replied:
Got it! I will ensure we maintain consistency with the Shadcn components and follow the standard practices to make future extensions easier. Every time I will check for updates before making custom modifications. Thanks You!

jonahtwalker commented on src/lib/components/settings-ui/notification-settings.svelte:
This file was moved to the `routes/settings/notifications` directory. It shouldn't be in the components folder as a Template or View.

rupalirajput2597 replied:
Got it ! Thank you!

jonahtwalker commented on src/lib/components/ui/sidebar/sidebar-trigger.svelte:
```diff
@@ -17,30 +17,22 @@
   const sidebar = useSidebar();
 </script>
 
-<Menu
```
@RaghvindYadav Please be sure that we are using the ready-made components from shad wherever possible We don't have a need for custom components yet, and all that was required here was to change the icon to a chevron.

RaghvindYadav replied:
I remember, I did change for a icon (hamburger), Okay I will never change ready made components.

jonahtwalker committed:
- removed `class` from DialogRoots in account settings

jonahtwalker commented on src/lib/utils/validation.ts:
While we should have a `utils` folder, this validation logic is probably most at home alongside the UpdatePasswordForm, when that is built. I'm imaging it's in a validation.ts in the same directory.

jonahtwalker replied:
I also don't know quite how this ties into our validation yet, or what the lib that we select may prefer. Zod? Tanstack Forms maybe? Superforms? I haven't looked into this stuff myself, yet.

rupalirajput2597 replied:
We need to figure it out!, I will also help you with this

jonahtwalker commented on src/routes/+layout.svelte:
```diff
@@ -6,11 +6,10 @@
 
 <div class="flex min-h-screen flex-col bg-background font-sans antialiased">
-  <Sidebar.Provider bind:open>
+  <Sidebar.Provider>
```
We are now correctly handling this in the AppSidebar itself.

rupalirajput2597 replied:
Yes, I checked it!

jonahtwalker commented on src/routes/faq/+page.svelte:
@rupalirajput2597 This came from your PR 👍🏼

jonahtwalker commented on src/routes/settings/+layout.svelte:
@rupalirajput2597 

I have refactored the settings pages/views. They are now all under the `routes/settings` directory. We will need to further refactor and modularize both the Account Settings when we do the Forms for making updates to the User with the node-client, too. I think it could be broken down into:
- AccountSettingsRow
- UpdateEmailDialog, UpdatePasswordDialog, UpdateUsernameDialog, etc.

I'm sure we can take it further, but even that would have all of these file sizes and be much more dry.

rupalirajput2597 replied:
Thanks for placing all settings under routes/settings makes things much more organized and I agree that breaking it down into AccountSettingsRow, UpdateEmailDialog, UpdatePasswordDialog, etc., will help keep things modular and maintainable.

jonahtwalker commented on src/routes/settings/+page.svelte:
```diff
@@ -1,50 +1,8 @@
 
-  // State management using Svelte 5 runes
-  let activeTab = $state('account'); // 'account' or 'notifications'
+  onMount(() => {
```
@rupalirajput2597 

`$effect` is meant to react to a mounted _update_, so I chose to use the dedicated `onMount` as that is all we need here. Until it's necessary, let's maintain that legibility and save `$effect`s for when we do genuinely require them, even if we can just use it without dependencies. `$effect(() =>)`

rupalirajput2597 replied:
Got it!, Thank you for the detailed explanation.

jonahtwalker commented on src/routes/settings/account/+page.svelte:
This is just cleanup, as these features are probably far out in Milestone 04 or 05.

rupalirajput2597 replied:
Cool, anyways these are planned for later milestones and keeping the code clean makes sense

jonahtwalker commented on src/routes/settings/notifications/+page.svelte:
Though not a priority right now, this file could definitely be broken down into a few pieces dedicated to the categories.

jonahtwalker commented on src/routes/signup/+page.svelte:
```diff
@@ -87,64 +66,6 @@
-      result.isValid = false;
-    }
-
-    if (email) {
```
This email validation logic has been dropped out of the password validation logic.

rupalirajput2597 replied:
Okay

rupalirajput2597 replied:
@jonahtwalker I noticed one more thing while testing your PR—we need to add validation for updating the email on the Account Settings screen as well. We can handle this as part of the 'Refactor Username Form' task or create a dedicated task for refactoring the 'Update Email' form. Let me know your thoughts!

<img width="893" alt="Screenshot 2025-03-18 at 6 03 54 PM" src="https://github.com/user-attachments/assets/133b8791-e374-4972-8814-c4ad7e57ca6e" />

jonahtwalker commented on src/svelte-coding-instructions.txt:
This is being deleted in favor of the LLM medium.txt in `docs`

jonahtwalker commented on tests/unit/nav-bar.test.ts:
```diff
@@ -21,22 +21,22 @@ describe('NavBar', () => {
 
-    await expect(logInButton).toBeVisible();
-    await expect(signUpButton).toBeVisible();
+    expect(logInButton).toBeVisible();
```
@jahanvi-baragaun I haven't found any of these awaits to be necessary.

jahanvi-baragaun replied:
yes. I tried without await and it worked. I will remove in my branch as I have other unit test files changes.

jonahtwalker reviewed (commented):
More to come, just commenting as we move along. 👍🏼

jonahtwalker committed:
- removed container class from the Footer child

jonahtwalker committed:
- navbar refactors
- the tests need to be revisited and completed

jonahtwalker commented:
There are two to-dos:
- [ ] fix and add some more tests for the navbar
- [x] the sidebar jumps open when you reduce the screen size from large > small. It needs it's state initialized differently, as the icon appears as a chevron in a mobile view while in what should be a collapsed state

jonahtwalker commented on src/lib/components/footer.svelte:
Removing the container class allows for the content to occupy the max width just as the Footer parent does.

rupalirajput2597 replied:
Ohh cool, Thank for letting me know!

jonahtwalker commented on src/lib/components/nav-bar.svelte:
This was moved to `components/nav-bar/nav-bar.svelte`, alongside the `avatar-menu.svelte`

jonahtwalker commented on src/lib/components/nav-bar/avatar-menu.svelte:
When the viewport width is less than medium, we collapse the NavBar menu items into this AvatarMenu. When unauthenticated, the AvatarMenu presents as an `...` dropdown button, as opposed to an additional sheet on the right side.

rupalirajput2597 replied:
@jonahtwalker Looks great! Just one concern ; in the unauthenticated state, the AvatarMenu has the 'Sign In' button at the bottom with a logout icon. It's an easy fix, but I wanted to point it out here for clarity. 

<img width="706" alt="Screenshot 2025-03-18 at 5 43 36 PM" src="https://github.com/user-attachments/assets/1b594c9e-8eb0-4496-a20d-a10404b84b99" />

jonahtwalker replied:
Thank you for pointing this out! Funny, I didn't notice that the example dropdown came with the wrong icon! Good catch. 👍🏼

jonahtwalker commented on src/lib/components/user-nav.svelte:
This was moved, and renamed to `avatar-menu.svelte`

rupalirajput2597 commented on src/routes/settings/account/+page.svelte:
```diff
@@ -8,7 +8,9 @@
   import DialogOverlayBlur from '$lib/components/ui/dialog/dialog-overlay-blur.svelte';
```
Hii @jonahtwalker We can remove the `DialogOverlayBlur` component since it is not an officially installed Shadcn file—it was generated by AI. It just used at one place in account-setting. So for simplicity, we should first delete this component and then re-install it  using:
```
npx shadcn-svelte@next add dialog
```
This will ensure consistency with Shadcn’s standard components as you mensioned in one of your comment.

rupalirajput2597 commented on src/routes/settings/account/+page.svelte:
```diff
@@ -8,7 +8,9 @@
   import DialogOverlayBlur from '$lib/components/ui/dialog/dialog-overlay-blur.svelte';
```
Hii @jonahtwalker We can remove the `DialogOverlayBlur` component since it is not an officially installed Shadcn file—it was generated by AI. It just used at one place in account-setting. So for simplicity, we should first delete this component and then re-install it  using:
```
npx shadcn-svelte@next add dialog
```
This will ensure consistency with Shadcn’s standard components as you mensioned in one of your comment.

jonahtwalker replied:
Thank you Rupali!! I didn't catch that, either

jahanvi-baragaun commented:
> There are two to-dos:
> 
> * [ ]  fix and add some more tests for the navbar
> * [ ]  the sidebar jumps open when you reduce the screen size from large > small. It needs it's state initialized differently, as the icon appears as a chevron in a mobile view while in what should be a collapsed state

Shall we add more tests later as in my branch I hv some changes to add svelte-i18n.

jonahtwalker committed:
- removed DialogOverlayBlur

jonahtwalker committed:
- added mode-watcher package
- refactored menu parts
- added LanguageButton
- added ThemeButton

jonahtwalker committed:
- fix sidebar trigger or condition

jonahtwalker committed:
formatting changes

jonahtwalker committed:
- added bg-node-client
- added MyUserContext
- added some documentation on the data layer
- added myusercontext to the navbar
- refactored validation.ts to be passwordHelpers
- more doc cleanup

jonahtwalker committed:
- signin form refactors

jonahtwalker commented:
**Please leave me a review of this PR.**

@RaghvindYadav 

This now contains the bg-node-client context. Please review the comments left by Holger in your PR, and close that when you're done there. New work should be based on this branch, which is intended to merge into `milestone-01` tomorrow, PST.

Please focus your work making the necessary updates that remain to be able to successfully sign up. If you have time, move on to signin. I have made major refactors to the SignIn page, please review those specifically as an example to implement signup similarly. Wherever possible, break down the file into components. It may be best if we have a file for each step in the process. There is a comment in the signin +page.svelte that has the latest `signup` shape. Lastly, please review [Holger's comment in slack from yesterday](https://baragaun.slack.com/archives/C084NTKGT26/p1742259116767239).

@rupalirajput2597

By starting to refactor the SignIn page, I have started to create the SignInForm. I think this is a good example to implement other forms as components in the settings tabs.

@jahanvi-baragaun 

I think it'd be best at this point, if possible - to separate the unit test updates and the localization stuff in your PR. I'd like to get both in and up to date, but they are going to need to be based on this branch - where the priority is working unit tests.

---

Until your day ends tomorrow, I am going to continue to work in this branch with new branches based off of this one to try and keep conflicts to a minimum over night.

jonahtwalker committed:
- updated signup shape

jonahtwalker committed:
- renamed context dir to contexts
- updated passwordHelpers in account settings and signup
- updated my-user-provider to use runes instead of snippets
- added myUserContext to navbar and avatarmenu to retrieve auth state

jonahtwalker committed:
- started refactoring the SignUpForm to use validated Forms from shadcn

rupalirajput2597 committed:
created update-username-input, account-setting-form, notification-setting-form (#49)

* refactored account settings into update-username-input, account-setting-form, notification-setting-form

RaghvindYadav committed:
story init (#54)

* initial storybook set up
* README formatting

---------

Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>

RaghvindYadav committed:
Integration of SignIn and SignUp with node client. Create mock my-user-context for storybook. (#55)

* signup and signin
* storybook update
* adds new authentication components
* updated node-client config
* added myUser context handling
* bg-node-client upgraded to 0.0.21

---------

Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>
Co-authored-by: shadowbrush <github@shadowbrush.com>

jonahtwalker committed:
test fixes, avatarmenu visibility broke

jonahtwalker commented:
This has a unit test that is failing as expected (avatarmenu visibility changed), and fails all storybook testing.

jonahtwalker committed:
fix avatarmenu visibility
